numpy/core/src/multiarray/convert_datatype.c

```diff
@@ -3293,8 +3296,10 @@ void_to_void_resolve_descriptors(
                 casting = NPY_NO_CASTING | _NPY_CAST_IS_VIEW;
             }
         }
-        NPY_CASTING field_casting = PyArray_GetCastSafety(
-                given_descrs[0]->subarray->base, given_descrs[1]->subarray->base, NULL);
+
+        PyArray_Descr *from_base = (from_sub == NULL) ? NULL : from_sub->base;
+        PyArray_Descr *to_base = (to_sub == NULL) ? NULL : to_sub->base;
+        NPY_CASTING field_casting = PyArray_GetCastSafety(from_base, to_base, NULL);
```

Comment:
This is the right idea, but should be `given_descrs[0]` or `given_descrs[1]` rather than `NULL`.  I guess the unstructured `void` case is missing (i.e. there must be a `V4` or similar involved here somehow), I don't know yet how or why `view` even triggers it...

For tests, it may be good to use also check with `np.can_cast` (`casting="unsafe"`).

Reply:
Ah, I had not realized that in your example `x.dtype` is `"V40"`, than it all adds up.

Reply:
Sure, that works without requiring any change to `PyArray_GetCastSafety` itself. Let me look into a test case and I will update